lib/silverpop/engage.rb

```diff
@@ -61,6 +61,12 @@ def logout
         @session_encoding = nil
       end
       success?
+
+    rescue => e
+      if e.message == 'Session has expired or is invalid'
+        @session_id = nil
+        @session_encoding = nil
+      end
```

Comment:
You'll want to `raise e` if the message doesn't match, otherwise we're suppressing all errors.